catalog/solr/catalog-solr-offline-gazetteer/src/main/java/ddf/catalog/solr/offlinegazetteer/GazetteerQueryOfflineSolr.java

```diff
@@ -0,0 +1,426 @@
+import org.apache.solr.client.solrj.util.ClientUtils;
+import org.apache.solr.common.SolrDocument;
+import org.codice.ddf.spatial.geocoding.GeoCodingConstants;
+import org.codice.ddf.spatial.geocoding.GeoEntry;
+import org.codice.ddf.spatial.geocoding.GeoEntryAttributes;
+import org.codice.ddf.spatial.geocoding.GeoEntryQueryException;
+import org.codice.ddf.spatial.geocoding.GeoEntryQueryable;
+import org.codice.ddf.spatial.geocoding.Suggestion;
+import org.codice.ddf.spatial.geocoding.context.NearbyLocation;
+import org.codice.solr.client.solrj.SolrClient;
+import org.codice.solr.factory.SolrClientFactory;
+import org.locationtech.jts.geom.Geometry;
+import org.locationtech.jts.geom.Point;
+import org.locationtech.jts.io.WKTReader;
+import org.locationtech.jts.io.WKTWriter;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class GazetteerQueryOfflineSolr implements GeoEntryQueryable {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(GazetteerQueryOfflineSolr.class);
+
+  private static final String CITY_SOLR_QUERY =
+      GeoCodingConstants.CITY_FEATURE_CODES
+          .stream()
+          .map(fc -> String.format("ext.feature-code_txt:%s", fc))
+          .collect(Collectors.joining(" OR ", "(", ")"));
+
+  private static final ThreadLocal<WKTWriter> WKT_WRITER_THREAD_LOCAL =
+      ThreadLocal.withInitial(WKTWriter::new);
+
+  private static final ThreadLocal<WKTReader> WKT_READER_THREAD_LOCAL =
+      ThreadLocal.withInitial(WKTReader::new);
+
+  protected static final String SUGGEST_Q = "suggest.q";
+  protected static final String SUGGEST_DICT = "suggest.dictionary";
+  protected static final String SUGGEST_PLACE_KEY = "suggestPlace";
+
+  public static final int MAX_RESULTS = 100;
+  public static final double KM_PER_DEGREE = 111.139;
+
+  private SolrClientFactory clientFactory;
+
+  private SolrClient client;
+
+  private final ExecutorService executor;
+
+  public GazetteerQueryOfflineSolr(
+      SolrClientFactory clientFactory, ExecutorService startupBuilderExecutor) {
+    this.clientFactory = clientFactory;
+    this.client = clientFactory.newClient(OfflineGazetteerPlugin.STANDALONE_GAZETTEER_CORE_NAME);
+    this.executor = startupBuilderExecutor;
+    executor.submit(
+        () -> {
+          RetryPolicy retryPolicy =
+              new RetryPolicy()
+                  .retryOn(Collections.singletonList(Exception.class))
+                  .withMaxDuration(20, TimeUnit.SECONDS)
+                  .withBackoff(100, 1_000, TimeUnit.MILLISECONDS);
+          SolrPingResponse ping =
+              Failsafe.with(retryPolicy)
+                  .onFailure(
+                      e ->
+                          LOGGER.error(
+                              "Could not get solrclient to start initial suggester build for {} core. Please try to start a build manually with the `build-suggester-index` karaf command or by sending a request to solr with the property `suggest.build=true`",
```

Comment:
This isn't actually a bundle breaking issue -- only building of the initial suggester. The bundle should continue on fine once the solr client connects and no restart would be necessary. So if we threw an exception and broke the bundle from starting it would have to manually be connected only at a time when solr was currently active. With the current way it will only not work while it can't connect to solr. Once connected it will resume working as expected.

Reply:
Perfect 👍  then let's leave it as-is.